core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -144,7 +150,13 @@ private static Map<String, String> loadEventListenerProperties(File configFile)
     public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
     {
         try (TimeStat.BlockTimer _ = queryCompletedTime.time()) {
+            if (concurrentQueryCompletedEvents.incrementAndGet() > maxConcurrentQueryCompletedEvents) {
```

Comment:
Do want the same logic for `queryCreated`? (but can keep a single config flag for both)

Reply:
`queryCreated` is actually blocking query from execution. Better be fast!